### ISSUE TITLE
Carousel With Min & Max Ranges: resize event is partially working

### ISSUE THREAD

**Issue by vlrprbttst** [woocommerce/FlexSlider#1452]
i have a carousel with 5 images and some breakpoints. here's my complete javascript

```
var $window = $(window),
    flexslider;

function getGridSize() {
    return ( (window.innerWidth || document.documentElement.clientWidth) < 500) ? 1 : ( (window.innerWidth || document.documentElement.clientWidth) < 890) ? 2 : ( (window.innerWidth || document.documentElement.clientWidth) < 1056) ? 3 : ( (window.innerWidth || document.documentElement.clientWidth) < 1920) ? 5 : 5;
}



$window.load(function() {
    $('.flexslider-cities').flexslider({
        animation: "slide",
        animationLoop: false,
        itemWidth: 180,
        itemMargin: 0,
        minItems: getGridSize(),
        maxItems: getGridSize()
    });
});

$window.resize(function() {
    var gridSize = getGridSize();

    $('.flexslider-cities').data('flexslider').vars.minItems = gridSize;
    $('.flexslider-cities').data('flexslider').vars.maxItems = gridSize;
});
```

i removed `syntaxHighlighter` is it was giving me an error in the console: why is that there anyway?

btw, the thing is actually working but i've noticed a problem when i resize my window: not every single px resizes the carousel the way it should. for example, i end up with this after a random resize:

![a](https://cloud.githubusercontent.com/assets/4832752/11064951/cbff8d2c-87c0-11e5-96ca-a0af46857d2c.png)

and if at this point I resize another few pixels, it will snap in place and return the correct result

![b](https://cloud.githubusercontent.com/assets/4832752/11064962/d99aaab6-87c0-11e5-9067-f0507b44b706.png)

this is not a big deal on desktop, but it is on mobile as if i flip the phone orientation (portrait to landscape or viceversa), the result is never right and it stays that way (as there's no way of resizing a mobile browser window any further).

i tried: 
- wrapping the whole flexslider init function into the window resize function to kickstart it from scratch on every px of window.resize event.
- wrapping the whole flexslider init function into a setInterval function to kickstart it from scratch every half second
- I removed  

```
$(function() {
    SyntaxHighlighter.all();
  });
```

as it was giving me an error in the console: why is that there anyway?
- i've corrected the function
  
  ```
  $window.resize(function() {
    var gridSize = getGridSize();
  
    flexslider.vars.minItems = gridSize;
    flexslider.vars.maxItems = gridSize;
  });
  ```
  
  as it was giving an error in the console (found it here on github or somewhere online)

i tried so many things and didn't come to a conclusion. any help or hack is greatly appreciated.

**Comment by theenoahmason** [comment]
Can confirm this issue with carousels.

**Comment by theenoahmason** [comment]
**Note:** not having console errors as mentioned in this thread, but I am getting unreliable slide sizing on resize when using *maxItems*

**Comment by matthiasm11** [comment]
Adding the function doMatch() works form me. Resizing window or rotating a device will recalculate the itemMargin, minItems and maxItems.

```
$window.resize(function() {
  var gridSize = getGridSize();

  flexslider.vars.minItems = gridSize;
  flexslider.vars.maxItems = gridSize;

  flexslider.doMath();
});
```

**Comment by matthiasm11** [comment]
Since the width of the items has been recalculated using doMath(), the height should also be recalculated.

Please make the function smoothHeight() public, or create a new API function:
`
slider.refresh = function() {
  slider.doMath();
  methods.smoothHeight();
};
`

**Comment by jeffikus** [comment]
Hi there, I released 2.7.0 last week, please let me know if the issue persists after updating.
